applications/FluidDynamicsApplication/custom_processes/surface_smoothing_process.cpp

```diff
@@ -0,0 +1,126 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Mohammad R. Hashemi
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "surface_smoothing_process.h"
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+SurfaceSmoothingProcess::SurfaceSmoothingProcess(
+    ModelPart& rModelPart,
+    TLinearSolver::Pointer plinear_solver)
+    : Process(),
+      mrModelPart(rModelPart)
+{
+    // Generate an auxilary model part and populate it by elements of type SurfaceSmoothingElement
+    CreateAuxModelPart();
+
+    auto p_builder_solver = Kratos::make_shared<ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> >(plinear_solver);
+
+    InitializeSolutionStrategy(plinear_solver, p_builder_solver);
+}
+
+void SurfaceSmoothingProcess::CreateAuxModelPart()
+{
+    Model& current_model = mrModelPart.GetModel();
+    if(current_model.HasModelPart( mAuxModelPartName ))
+        current_model.DeleteModelPart( mAuxModelPartName );
+
+    // Adding DISTANCE to the solution variables is not needed if it is already a solution variable of the problem
+    mrModelPart.AddNodalSolutionStepVariable(DISTANCE);
+
+    // Ensure that the nodes have distance as a DOF
+    VariableUtils().AddDof<Variable<double> >(DISTANCE, mrModelPart);
+
+    // Generate AuxModelPart
+    ModelPart& r_smoothing_model_part = current_model.CreateModelPart( mAuxModelPartName );
+
+    Element::Pointer p_smoothing_element = Kratos::make_intrusive<SurfaceSmoothingElement>();
+
+    ConnectivityPreserveModeler modeler;
+    modeler.GenerateModelPart(mrModelPart, r_smoothing_model_part, *p_smoothing_element);
+
+    const double delta_time = mrModelPart.pGetProcessInfo()->GetValue(DELTA_TIME);
+    r_smoothing_model_part.pGetProcessInfo()->SetValue(DELTA_TIME, delta_time);
```

Comment:
The process info is shared between the origin and the destination model parts, so this shouldn't be needed.

Reply:
In the previous Kratos release (which I use in my own branch) this was necessary. It was strange to me but I found it out eventually as the source of error.

Reply:
do you create a new ModelPart?

Reply:
Yes, I need to utilize a specific element to solve the diffusion equation.

Reply:
I checked the implementation and the auxiliary model part points to the origin model part container... If I'm not wrong this shouldn't be any longer required :confused:

Reply:
@rubenzorrilla you are right
https://github.com/KratosMultiphysics/Kratos/blob/1c6ca9d6fae7c1fd1728c8ce4f590b0ccfd0871b/kratos/sources/connectivity_preserve_modeler.cpp#L141

Maybe I have made a mistake somewhere else! I will check.